frontend/appComponents/Header/components/SignInLinks.js

```diff
@@ -1,13 +1,22 @@
+import { withRouter } from 'react-router-dom';
+
+@withRouter
+
 class SignInLinks extends React.Component {
+  
+  redirectBackTo = () => {
+    sessionStorage.setItem("lastpage",this.props.location.pathname);
```

Comment:
Do you use eslint? Please install eslint to your text editor, it should highlight nonstandard syntax for you (e.g. there should always be a space after `,`)